tests/unit/core/handler-test.ts

```diff
@@ -244,15 +244,108 @@ module('Unit | core/handler', function (hooks) {
     assert.equal(handler.currentPage, 1);
   });
 
-  test('Handler#toggleSelectAll', function (assert: Assert) {
-    const handler = new TableHandler(getContext(), this.tableManager, this.rowsFetcher);
-    assert.deepEqual(handler.selection, []);
+  module('Handler#toggleSelectAll', () => {
+    test('it selects all the loaded rows', async function (assert: Assert) {
+      const handler = new TableHandler(getContext(), this.tableManager, this.rowsFetcher);
+      assert.deepEqual(handler.selection, []);
 
-    handler.toggleSelectAll(true);
-    assert.deepEqual(handler.selection, 'all');
+      await handler.fetchRows();
+      handler.toggleSelectAll(true);
+      assert.equal(handler.selection.length, 2);
+    });
 
-    handler.toggleSelectAll(false);
+    test('it selects all the rows', async function (assert: Assert) {
+      this.rowsFetcher = new AllRowsFetcher();
+      const handler = new TableHandler(getContext(), this.tableManager, this.rowsFetcher);
+      assert.deepEqual(handler.selection, []);
+
+      await handler.fetchRows();
+      handler.toggleSelectAll(true);
+      assert.equal(handler.selection, 'all');
+    });
+
+    test('it clears the selected rows', async function (assert: Assert) {
+      const handler = new TableHandler(getContext(), this.tableManager, this.rowsFetcher);
+      assert.deepEqual(handler.selection, []);
+      await handler.fetchRows();
+
+      handler.toggleSelectAll(true);
+      handler.toggleSelectAll(false);
+
+      assert.deepEqual(handler.selection, []);
+    });
+  });
+
+  test('Handler#selectAllGlobal', async function (assert) {
+    const handler = new TableHandler(getContext(), this.tableManager, this.rowsFetcher);
+    handler.selection = [
+      {
+        influencerId: 42,
+        recordId: 12,
+        record_id: 12,
+        holderId: 57,
+        holderType: 'list',
+        foo: 'ekip',
+        bar: 'hello',
+        total: 123,
+        date: 1643386394
+      }
+    ];
+    handler.exclusion = [
+      {
+        influencerId: 42,
+        recordId: 12,
+        record_id: 12,
+        holderId: 57,
+        holderType: 'list',
+        foo: 'ekip',
+        bar: 'hello',
+        total: 123,
+        date: 1643386394
+      }
+    ];
```

Comment:
Same lenghty definition as below.
how about a simple function that returns the row object since it's the same in all 4 cases ?